packages/account-v2/src/containers/POAForm/AddressDetailsForm.tsx

```diff
@@ -0,0 +1,127 @@
+import { useHistory } from 'react-router-dom';
+import { useDocumentUpload, useSettings } from '@deriv/api';
+import { useBreakpoint } from '@deriv/quill-design';
+import { StandaloneXmarkBoldIcon } from '@deriv/quill-icons';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import { InlineMessage } from '../../components/base/InlineMessage';
+import { AddressFields } from '../../modules/AddressFields';
+import DocumentSubmission from './DocumentSubmission';
+import { POAError } from './Status';
+
+type TAddressDetails = {
+    addressCity: string;
+    addressLine1: string;
+    addressLine2: string;
+    addressPostcode: string;
+    addressState: string;
+    document?: File;
+};
+
+type TAddressDetailsForm = {
+    resubmitting?: boolean;
+};
+
+export const AddressDetailsForm = ({ resubmitting }: TAddressDetailsForm) => {
+    const { data: settings, error: fetchError, mutation, update: updateSettings } = useSettings();
+    const { error: documentUploadError, isLoading: isDocumentUploading, upload } = useDocumentUpload();
+    const { isMobile } = useBreakpoint();
+    const history = useHistory();
+
+    const { error: settingsUpdateError, isLoading: isSettingsUpdating } = mutation;
+
+    const handleFormSubmit = async (values: TAddressDetails) => {
+        updateSettings({
+            address_city: values.addressCity,
+            address_line_1: values.addressLine1,
+            address_line_2: values.addressLine2,
+            address_postcode: values.addressPostcode,
+            address_state: values.addressState,
+        });
+
+        // upload file
+        return upload({
+            document_issuing_country: settings?.country_code ?? undefined,
+            document_type: 'proofaddress',
+            file: values.document,
+        });
+    };
+
+    const { address_city, address_line_1, address_line_2, address_postcode, address_state } = settings;
+
+    const initialValues: TAddressDetails = {
+        addressCity: address_city ?? '',
+        addressLine1: address_line_1 ?? '',
+        addressLine2: address_line_2 ?? '',
+        addressPostcode: address_postcode ?? '',
+        addressState: address_state ?? '',
+    };
+
+    if (fetchError) {
+        return <POAError error_message={fetchError.error.message} />;
+    }
+
+    const formSubmitError = settingsUpdateError?.error.message || documentUploadError?.error.message;
+
+    return (
+        <Fragment>
+            {isMobile && (
+                <div className='grid grid-cols-[auto_25px] items-center pb-300 mb-500 border-solid border-b-75 border-solid-grey-2'>
+                    <Text align='center' size='lg' weight='bold'>
+                        Proof of address
+                    </Text>
+                    <StandaloneXmarkBoldIcon
+                        iconSize='md'
+                        onClick={() => {
+                            history.push('/account-v2/');
```

Comment:
Can we replace the string with value from `'@deriv/shared'`

Reply:
are we adding @deriv/shared as dependency in account-v2 ?

Reply:
Created a routes constant file for this.